drivers/input/touchscreen/ili210x.c

```diff
@@ -1001,12 +1027,18 @@ static int ili210x_i2c_probe(struct i2c_client *client)
 		return error;
 	}
 
-	error = devm_request_threaded_irq(dev, client->irq, NULL, ili210x_irq,
-					  IRQF_ONESHOT, client->name, priv);
-	if (error) {
-		dev_err(dev, "Unable to request touchscreen IRQ, err: %d\n",
-			error);
-		return error;
+	if (client->irq) {
+		error = devm_request_threaded_irq(dev, client->irq, NULL, ili210x_irq,
+					IRQF_ONESHOT, client->name, priv);
+		if (error) {
+			dev_err(dev, "Unable to request touchscreen IRQ, err: %d\n",error);
+			return error;
+		}
+	}
+	else{
+		timer_setup(&priv->poll_timer, ili210x_poll_timer_callback, 0);
+		mod_timer(&priv->poll_timer, jiffies + msecs_to_jiffies(ILI2XXX_POLL_PERIOD));
+		INIT_WORK(&priv->poll_work, ili210x_poll_work);
```

Comment:
You don't cancel the workqueue or delete the timer on remove.
See https://github.com/raspberrypi/linux/commit/818bb6d49db23300ab816fe6e1e71b6387f79d96 for doing the same on edt-ft5x06(*).

In this case it probably wants to go into ili210x_stop though as they've avoided adding a remove function.

(*) Looking at it, I think the cancel_work_sync should come first.
Comparing to [tc358743](https://elixir.bootlin.com/linux/v6.12.1/source/drivers/media/i2c/tc358743.c#L2186) which also has a polling option, that uses slightly different semantics again - there never is one clear option, and it needs careful thought over what triggers what.

Reply:
Thank you very much for your valuable guidance on maintaining coding standards, especially for patches intended for upstream submission.I will promptly address the issues identified by checkpatch and correct the coding style problem you pointed out. Once the updates are complete, I will submit the revised version for further review.